src/Sugar.mli

```diff
@@ -0,0 +1,169 @@
+    arguments and the body of the function [exp]. [cmts] are relocated to
+    the arguments or the function body to have more precise positions.
+    [will_keep_first_ast_node] is set by default, otherwise the [exp] is
+    returned without modification. *)
+
+val sugar_infix :
+     Cmts.t
+  -> Ast.prec option
+  -> expression Ast.xt
+  -> (expression Ast.xt option * (arg_label * expression Ast.xt) list) list
+(** [sugar_infix cmts prec exp] returns the infix operator and the list of
+    operands applied to this operator from expression [exp]. [prec] is the
+    precedence of the infix operator. [cmts] are relocated to the operand to
+    have more precise positions. *)
+
+val sugar_list_pat :
+     Cmts.t
+  -> pattern
+  -> ((Warnings.loc list * pattern Ast.xt) list * Warnings.loc) option
+(** [sugar_list_pat cmts pat] returns a list of patterns if [pat] is a
+    pattern corresponding to a list (empty list or (::) application). [cmts]
+    are relocated to the list elements to have more precise positions. *)
+
+val sugar_list_exp :
+     Cmts.t
+  -> expression
+  -> ((Warnings.loc list * expression Ast.xt) list * Warnings.loc) option
+(** [sugar_list_exp cmts exp] returns a list of expressions if [exp] is an
+    expression corresponding to a list (empty list or (::) application).
+    [cmts] are relocated to the list elements to have more positions. *)
+
+val sugar_infix_cons :
+  expression Ast.xt -> (Warnings.loc list * expression Ast.xt) list
+(** [sugar_infix_cons exp] returns a list of expressions if [exp] is an
+    expression corresponding to a list ((::) application). *)
+
+val sugar_ite :
+     Cmts.t
+  -> expression Ast.xt
+  -> (expression Ast.xt option * expression Ast.xt * attributes) list
+(** [sugar_ite cmts exp] returns a list of conditional expressions from
+    cascading if-then-else expressions, e.g.:
+
+    {[
+        if c1 then e1
+        else if c2 then e2
+        else e3
+    ]}
+
+    will return the following list:
+    [(Some c1, e1); (Some c2, e2); (None, e3)] (attributes are voluntarily
+    omitted. [cmts] are relocated to the sub-expressions to have more
+    precise positions. *)
+
+val sugar_sequence : Cmts.t -> expression Ast.xt -> expression Ast.xt list
+(** [sugar_sequence cmts exp] returns the list of expressions from a
+    sequence of expressions [exp]. [cmts] are relocated to the
+    sub-expressions to have more precise positions. *)
+
+val sugar_functor_type :
+     Cmts.t
+  -> for_functor_kw:bool
+  -> module_type Ast.xt
+  -> (label loc * module_type Ast.xt option) list * module_type Ast.xt
+(** [sugar_functor_type cmts for_functor_kw m] returns the list of module
+    types applied to the functor of module type [m]. [for_functor_kw]
+    indicates if the keyword [functor] is used. The sugar is different when
+    used with the [functor] keyword. The syntax M(A : A)(B : B) cannot
+    handle [_] as module name. [cmts] are relocated to the functor arguments
```

Comment:
The comment about `_` as a module name is really an implementation-explanation comment and should be moved back.